src/realm/array.cpp

```diff
@@ -358,7 +314,12 @@ void Array::preset(int64_t min, int64_t max, size_t num_items)
 }
 
 
-void Array::destroy_children(size_t offset) noexcept
+void Array::destroy_children() noexcept
+{
+    destroy_some_children(0);
+}
+
+void Array::destroy_some_children(size_t offset) noexcept
```

Comment:
Not a great name. To me it indicates that not all elements are guaranteed to be destroyed, just "whatever the system feels like". What's wrong with `destroy_children(size_t offset = 0)`?